10up/ElasticPress
# Add new filter ep_skip_autosave_sync and enable autosave sync for protected_content feature

### Description of the Change

Problem: When a user has the `protected_content` feature enabled, and they work on a draft but they never hit "Save draft" and instead, rely on the WordPress autosave, it won't show up in the Draft post listing screen (/wp-admin/edit.php?post_status=draft&post_type=post):

<img width="1489" alt="Screen Shot 2022-06-24 at 1 43 00 PM" src="https://user-images.githubusercontent.com/16962021/175656754-625f8270-aa5c-474c-8d79-2c81eacdc387.png">

This is because it never gets synced, as we disable it here:

https://github.com/10up/ElasticPress/blob/aaf2a3342b61c5df958f874cfa1fb434c244201f/includes/classes/Indexable/Post/SyncManager.php#L284-L289

This PR does two things:
- Introduces the `ep_skip_autosave_sync` filter 
- Returns false for the aforementioned filter when the `protected_content` feature is enabled

### Alternate Designs

I'm not sure if it would be beneficial to pass in the `__FUNCTION__` for the second parameter during `apply_filters()` if one wanted to customize disabling the autosave sync on function-basis, since we have `action_set_object_terms`, `action_sync_on_update`, etc. to make it more granular:

```
		if ( apply_filters( 'ep_disable_autosave_sync', true, __FUNCTION__ ) ) {
			if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
				// Bypass saving if doing autosave
				// @codeCoverageIgnoreStart
				return;
				// @codeCoverageIgnoreEnd
			}
		}
```

Curious on your thoughts @felipeelia for that.

### Possible Drawbacks

N/A.

### Verification Process

1) Enable the `protected_content` feature
2) Start drafting a new post and wait for WordPress to autosave
3) Go to the Draft post listing screen at `/wp-admin/edit.php?post_status=draft&post_type=post` and expect to see the autosaved draft

### Checklist:






### Changelog Entry

Fixed: autosaved drafts not showing up in draft post listing when using protected_content

### Credits


Props @rebeccahum
